### PR TITLE
VAVFA-8986: Add facility expandable alert component.

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
Adds expandable alert for facility covid status usage. See:
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/8986
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/8981
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/8982
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/8983

Sibling pr for content-build vamc expandable widget needed for functionality: https://github.com/department-of-veterans-affairs/content-build/pull/1128

Tugboat data endpoint to use for testing: https://pr9028-sjpg4ebinnl4eooevngpwufok19n4akk.ci.cms.va.gov/

Adding @maxx1128 & @swirtSJW for situational awareness.

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/8986

## Testing done
Visual / eslint

## Screenshots
image

image

image

image

Screen Shot 2022-05-16 at 1 48 09 PM

Screen Shot 2022-05-16 at 11 12 34 AM

Screen Shot 2022-05-16 at 11 13 18 AM

## Acceptance criteria
- [ ] run `yarn watch` (in vets-website)
- [ ] Checkout https://github.com/department-of-veterans-affairs/content-build/pull/1128
 and in content-build, run `yarn preview --drupal-address=https://pr9028-sjpg4ebinnl4eooevngpwufok19n4akk.ci.cms.va.gov/ --drupal-user=ethan.teague --drupal-password=drupal8`
- [ ] Visually verify presence and functionality of expandable/collapsible covid status alerts when visiting http://localhost:3002/preview?nodeId=90
- [ ] - [ ] Visually verify presence and functionality of expandable/collapsible covid status alerts when visiting http://localhost:3002/preview?nodeId=1010
- [ ] - [ ] Visually verify presence and functionality of expandable/collapsible covid status alerts when visiting http://localhost:3002/preview?nodeId=2802

### CODE REVIEW

**Review by bkjohnson — CHANGES_REQUESTED** [review]
I think we should hold off on this - [`<va-alert-expandable>`](https://design.va.gov/storybook/?path=/docs/components-va-alert-expandable--default)  will be published in the next few days.

**Comment by ethanteague** [comment]
> I think we should hold off on this - [`<va-alert-expandable>`](https://design.va.gov/storybook/?path=/docs/components-va-alert-expandable--default) will be published in the next few days.
Not sure that we can, @bkjohnson - we need to release VAMC covid status alerts this week. Kind of a 911 situation. Looping in @kevwalsh @swirtSJW @maxx1128 @davidconlon @mmiddaugh @cjterdi for situational awareness.

**Review by it-harrison — APPROVED** [review]

**Comment by swirtSJW** [comment]
This was just put on hold for a week due to people ops reasons.    Also converting this to draft because it will break things since the field is not on prod.

File: src/applications/static-pages/facilities/buildVamcOperatingStatusProps.js
Comment (swirtSJW) [inline src/applications/static-pages/facilities/buildVamcOperatingStatusProps.js:10]: Should map to cases of the field_status_id to the values identified in the storybook component.   That way it is not specific to operating statuses which are not related.

File: src/applications/static-pages/facilities/buildVamcOperatingStatusProps.js
Comment (swirtSJW) [inline src/applications/static-pages/facilities/buildVamcOperatingStatusProps.js:1]: Naming on this is confusing.  It is not VamcOperatingstatus.     If anything it is supplemental status. ... but going even bigger it is alert expandable

File: src/applications/static-pages/facilities/ExpandableVamcOperatingStatus.jsx
Comment (swirtSJW) [inline src/applications/static-pages/facilities/ExpandableVamcOperatingStatus.jsx:1]: filename and location seems misnamed.    THis is supplemental status not operating status and going one level up it is really the pattern for alert expandabile.

File: src/applications/static-pages/facilities/ExpandableVamcOperatingStatus.jsx
Comment (swirtSJW) [inline src/applications/static-pages/facilities/ExpandableVamcOperatingStatus.jsx:39]: Can we go more genateric? alert expandible   from storybook?

File: src/applications/static-pages/facilities/ExpandableVamcOperatingStatus.jsx
Comment (swirtSJW) [inline src/applications/static-pages/facilities/ExpandableVamcOperatingStatus.jsx:42]: supplemental status or more generic as alert expandable?

File: src/applications/static-pages/facilities/ExpandableVamcOperatingStatus.jsx
Comment (swirtSJW) [inline src/applications/static-pages/facilities/ExpandableVamcOperatingStatus.jsx:64]: Can we make it less facility and status specific?

File: src/applications/static-pages/facilities/ExpandableVamcOperatingStatus.jsx
Comment (swirtSJW) [inline src/applications/static-pages/facilities/ExpandableVamcOperatingStatus.jsx:75]: supplemental status or alert expandable?

File: src/applications/static-pages/facilities/ExpandableVamcOperatingStatus.jsx
Comment (swirtSJW) [inline src/applications/static-pages/facilities/ExpandableVamcOperatingStatus.jsx:82]: naming?

File: src/applications/static-pages/facilities/buildVamcOperatingStatusProps.js
Comment (swirtSJW) [inline src/applications/static-pages/facilities/buildVamcOperatingStatusProps.js:25]: less specific more like heading and body?

File: src/applications/static-pages/facilities/createExpandableVamcOperatingStatus.jsx
Comment (swirtSJW) [inline src/applications/static-pages/facilities/createExpandableVamcOperatingStatus.jsx:3]: Same questions about generizizing the names used.

**Review by swirtSJW — DISMISSED** [review]
Mainly stuff we talked about on the call... and a little bit more pondering on naming. to make it either about supplemental status OR better yet alert expandable (generic)

**Comment by swirtSJW** [comment]
> I think we should hold off on this - [`<va-alert-expandable>`](https://design.va.gov/storybook/?path=/docs/components-va-alert-expandable--default) will be published in the next few days.

Thanks for raising this.   We are under a tight deadline for the getting this out to coincide with a VAMC  operations rollout timing that goes well beyond the website..    @ethanteague  and @maxx1128  are going to rework this to become that component.   Yes the implementation may happen before the design component is official, but we can rework this if needed to align with the official design when it happens.   Any guidance you can give along the way is much appreciated, but for timing the rework of this that is inbound within hours, needs to go out preferably today 5/12 or at the latests tomorrow.


This component is going to be moved to src/platform/site-wide/alerts/components and renamed.

**Review by bkjohnson — APPROVED** [review]
I'm not seeing the option to dismiss my review, but I'm approving so that I'm not blocking whatever feature flag changes need to happen.

File: src/applications/facility-locator/components/search-results-items/common/LocationCovidStatus.jsx
Comment (maxx1128) [reply src/applications/facility-locator/components/search-results-items/common/LocationCovidStatus.jsx:1]: Like `LocationOperationStatus.jsx`, this is disabled so the label can have a `tabIndex` value and be more accessible to keyboard users.

**Review by swirtSJW — APPROVED** [review]
Code and screencaps look good.

**Pull request merged**
